staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go

```diff
@@ -84,17 +84,33 @@ func TestApplyExtraArgsFail(t *testing.T) {
 	f := cmdtesting.NewTestFactory()
 	defer f.Cleanup()
 
-	c := NewCmdApply("kubectl", f, genericclioptions.NewTestIOStreamsDiscard())
-	if validateApplyArgs(c, []string{"rc"}) == nil {
-		t.Fatalf("unexpected non-error")
-	}
+	cmd := &cobra.Command{}
+	flags := NewApplyFlags(genericclioptions.NewTestIOStreamsDiscard())
+	flags.AddFlags(cmd)
+	_, err := flags.ToOptions(f, cmd, "kubectl", []string{"rc"})
+	require.EqualError(t, err, "Unexpected args: [rc]\nSee ' -h' for help and examples")
```

Comment:
The change above is not at all related to this PR. I was searching the file for examples of how to test flag exclusion, and this one caught my eye. In reading it, I realized it doesn't prove anything! It was just testing that the helper (which I deleted below) returns an error, which it always does. Since `ToOptions` seems responsible for extra args errors in the real command, I wired that up in the test.

Reply:
Nit: I prefer not to test exact error messages, this saves time when they change during other updates. I'd focus only on testing substrings which confirm what you're testing.

Reply:
I like that testing the whole message validates the real user experience we're providing on errors, and catches things like redundancy in the wrapping. But it's a tradeoff for sure, and in particular I buy the argument that the 'see -h' part isn't useful here. 😄